File: python/31_count_above_average.py
"""Contar elementos mayores que la media — Fácil · Arrays

Cuenta los elementos estrictamente mayores que la media; una lista vacía
devuelve 0.

Fuente: https://coding-challenges.dev/problems/python-contar-elementos-mayor-media
"""


def count_above_average(numbers: list[float]) -> int:
    if not numbers:
        return 0
    mean = sum(numbers) / len(numbers)
    return sum(number > mean for number in numbers)

File: typescript/31-count-above-average.ts
// Contar elementos mayores que la media — Fácil · Arrays
// Cuenta los elementos estrictamente mayores que la media; [] devuelve 0.
// Fuente: https://coding-challenges.dev/problems/typescript-contar-elementos-mayor-media

export function countAboveAverage(numbers: number[]): number {
  if (numbers.length === 0) return 0;
  const average = numbers.reduce((sum, number) => sum + number, 0) / numbers.length;
  return numbers.filter((number) => number > average).length;
}
